src/index.mjs

```diff
@@ -27,78 +35,162 @@ function directoryHasPackageJson(dir) {
+		case 'yarn':
+			return `${pkgManager} dlx`;
+		default:
+			throw new Error(`"${pkgManager}" package manager is not supported`);
+	}
+}
+
+/**
+ * @param {string} errorMessage
+ */
+function cancelSetup(errorMessage = '') {
+	console.error(
+		stripIndent(`
+			${picocolors.red(picocolors.bold('Setup canceled!'))}
+		`),
+	);
+
+	if (errorMessage) {
+		console.error(
+			stripIndent(`${errorMessage}
+			`),
+		);
+	}
+
+	process.exit(1);
+}
+
 /**
  * @param {string} pkgManager
  */
-function getInstallCommand(pkgManager) {
-	return pkgManager === 'npm' ? 'install' : 'add';
+async function showPrompt(pkgManager) {
+	console.info(
+		stripIndent(`
+			We'll create a '${DEFAULT_CONFIG_FILE}' file containing:
+		`),
+	);
+
+	console.info(
+		picocolors.dim(
+			DEFAULT_CONFIG_CONTENT.split('\n')
+				.map((line) => `  ${line}`)
+				.join('\n'),
+		),
+	);
+
+	console.info(
+		stripIndent(`
+			Then add the related dependencies using:
+
+			  ${picocolors.dim(`${pkgManager} ${ADD_COMMAND}`)}
+		`),
+	);
+
+	let proceed;
+
+	try {
+		proceed = await confirm({
+			message: 'Continue?',
+		});
+	} catch (error) {
+		if (error instanceof Error && error.name === 'ExitPromptError') {
+			// silence
+		} else {
+			throw error;
+		}
+	}
+
+	if (!proceed) {
+		cancelSetup();
+	}
 }
 
 /**
  * @param {string} cwd
  * @param {string} pkgManager
  */
-function createConfig(cwd, pkgManager) {
+async function createConfig(cwd, pkgManager) {
 	const spinner = ora('Creating config...').start();
-	const existingConfig = getExistingConfigInDirectory();
+	const existingConfig = await getExistingConfigInDirectory();
 
 	if (existingConfig !== null) {
 		const basename = path.basename(existingConfig.filepath);
 		const failureMessage =
 			basename === 'package.json'
-				? 'The "stylelint" config in "package.json" already exists.'
-				: `The "${basename}" config already exists.`;
+				? 'A "stylelint" config in "package.json" already exists.'
+				: `A "${basename}" config already exists.`;
 
-		spinner.fail(`Failed to create config:\n${failureMessage} Remove it and then try again.`);
-		process.exit(1);
+		spinner.fail();
+		cancelSetup(`${failureMessage} Remove it and then try again.`);
 	}
 
 	if (!directoryHasPackageJson(cwd)) {
-		spinner.fail(
-			`Failed to create config:\npackage.json was not found. Run "${pkgManager} init" and then try again.`,
-		);
-		process.exit(1);
+		spinner.fail();
+		cancelSetup(`A "package.json" was not found. Run "${pkgManager} init" and then try again.`);
 	}
 
 	try {
-		fs.writeFileSync(DEFAULT_CONFIG_FILE, '{ "extends": ["stylelint-config-standard"] }');
+		fs.writeFileSync(DEFAULT_CONFIG_FILE, DEFAULT_CONFIG_CONTENT);
 	} catch (error) {
-		spinner.fail(`Failed to create config:\n${error}`);
-		process.exit(1);
+		spinner.fail();
+		cancelSetup(error instanceof Error ? error.message : String(error));
 	}
 
-	spinner.succeed(`Created ${DEFAULT_CONFIG_FILE}.`);
+	spinner.succeed(`Created ${DEFAULT_CONFIG_FILE}`);
```

Comment:
[suggestion] How about emphasizing the file name, as well?


```suggestion
	spinner.succeed(`Created ${picocolors.greenBright(DEFAULT_CONFIG_FILE)}`);
```

Reply:
To reduce redundancy, I've pushed a commit (https://github.com/stylelint/create-stylelint/pull/105/commits/0085bf1e77b6e32b53393815c5c1d4387c2be9e4) that goes the other way i.e. making the succeed-messages succinct as the context is just above them.

I'm also keen to avoid too many colours, and would like to reserve them for a couple of key bits.